src/main/java/net/tropicraft/core/common/dimension/feature/HomeTreeBranchFeature.java

```diff
@@ -0,0 +1,147 @@
+package net.tropicraft.core.common.dimension.feature;
+
+import com.mojang.datafixers.Dynamic;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.VineBlock;
+import net.minecraft.state.BooleanProperty;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.gen.ChunkGenerator;
+import net.minecraft.world.gen.GenerationSettings;
+import net.minecraft.world.gen.feature.Feature;
+import net.tropicraft.core.common.block.TropicraftBlocks;
+import net.tropicraft.core.common.dimension.feature.config.HomeTreeBranchConfig;
+
+import java.util.Random;
+import java.util.function.Function;
+
+public class HomeTreeBranchFeature<T extends HomeTreeBranchConfig> extends Feature<T> {
+    private static final byte[] OTHER_COORD_PAIRS = {
+        2, 0, 0, 1, 2, 1
+    };
+
+    public HomeTreeBranchFeature(Function<Dynamic<?>, ? extends T> configSerializer) {
+        super(configSerializer);
+    }
+
+    @Override
+    public boolean place(IWorld world, ChunkGenerator<? extends GenerationSettings> generator, Random rand, BlockPos pos, T config) {
+        final int branchLength = rand.nextInt(10) + 15;
+        // TODO make configurable
+        int branchX1 = pos.getX();
+        int branchZ1 = pos.getZ();
+        final double minAngle = Math.toRadians(config.minAngle);
+        final double maxAngle = Math.toRadians(config.maxAngle);
+        double angle = minAngle + rand.nextFloat() * (maxAngle - minAngle);
+        branchX1 = (int)(Math.sin(angle) + branchX1);
```

Comment:
I'm not sure this math is right, why would you add the angle components to the starting position? Shouldn't it always *start* at the passed in position?

Reply:
changed